Realm/RLMSyncUser.mm

```diff
@@ -383,6 +368,65 @@ - (void)revokePermission:(RLMSyncPermission *)permission callback:(RLMPermission
                                    *_configMaker);
 }
 
+- (void)createOfferForRealmAtURL:(NSURL *)url
+                     accessLevel:(RLMSyncAccessLevel)accessLevel
+                      expiration:(NSDate *)expirationDate
+                        callback:(RLMPermissionOfferStatusBlock)callback {
+    if (!_user || _user->state() == SyncUser::State::Error) {
+        callback(nil, make_permission_error_change(@"A permission offer cannot be created using an invalid user."));
+        return;
+    }
+    auto cb = [callback](util::Optional<std::string> token, std::exception_ptr ptr) {
+        if (ptr) {
+            NSError *error = translateSyncExceptionPtrToError(std::move(ptr), RLMPermissionActionTypeOffer);
+            REALM_ASSERT(error);
+            callback(nil, error);
+        } else {
+            REALM_ASSERT_DEBUG(token != nullptr);
```

Comment:
`token` is an `Optional<std::string>` so comparing against `nullptr` seems odd. This can just be `REALM_ASSERT_DEBUG(token)`.